app/models/user.rb

```diff
@@ -351,4 +251,132 @@ def pcode3
   def pcode4
     school.sierra_code
   end
+
+
+  # Another piece of code has determined this user is all set to be called
+  # complete and done.  Note: Usually, expect the next step to be the calling of
+  # Patron Service to create the user in Sierra.
+  def save_as_complete
+    self.status = STATUS_LABELS['complete']
+    self.save
+  end
+
+
+  # If the user's barcode is not yet finalized, then set its status to
+  # 'pending' and save.  In the future, there may be other conditions that
+  # could set the user to "pending", and we'll be checking for those here, as well.
+  def save_as_pending
+    # do we need to fill in a provisional barcode?
+    unless self.barcode.present?
+      self.barcode = self.assign_barcode
+    end
+
+    self.status = STATUS_LABELS['barcode_pending']
+    self.save
+  end
+
+
+  # ################ THE BARCODES SECTION! ################
+
+  def assign_barcode
+    LogWrapper.log('DEBUG', {
+       'message' => "Begin assigning barcode to #{self.email}",
+       'method' => "assign_barcode",
+       'status' => "start",
+       'user' => {email: self.email}
+      })
+
+    last_user_barcode = User.where('barcode < 27777099999999').order(:barcode).last.barcode
+    self.assign_attributes({ barcode: last_user_barcode + 1})
+
+    LogWrapper.log('DEBUG', {
+       'message' => "Barcode has been assigned to #{self.email}",
+       'method' => "assign_barcode",
+       'status' => "end",
+       'barcode' => "#{self.barcode}",
+       'user' => {email: self.email}
+      })
+    return self.barcode
+  end
+
+
+  def barcode_found_in_sierra
```

Comment:
Is this still needed? Looks like you don't call it anymore, unless I'm missing something.

Reply:
It will be needed in the future, yes.

Reply:
Gotcha. 👍